internal/apigw/apigw_method.go

```diff
@@ -326,6 +324,135 @@ func (module *Module) generateFieldConverter(method pgs.Method, edgeNumber proto
 	}
 }
 
+func (module *Module) generateNestedProtoMessageOutput(idx int, edgeNumbers []protopack.Number, outputName string) string {
+	var inputName string
+
+	// The first edge number is the last step and writes to the output instead of an array
+	if idx == 0 {
+		inputName = outputName
+	} else {
+		inputName = fmt.Sprintf("arr[%d]", idx+1)
+	}
+
+	message, err := templateExecToString("protopack_message.tmpl", &protopackMessageContext{
+		InputName: inputName,
+		Number:    edgeNumbers[idx],
+		// We need to -2 because the last level is the initializer and the second to last level is the base case
+		Index: len(edgeNumbers) - idx - 2,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Base case
+	if idx == len(edgeNumbers)-2 {
+		return message
+	}
+	return fmt.Sprintf("%s\n%s", module.generateNestedProtoMessageOutput(idx+1, edgeNumbers, outputName), message)
+}
+
+func (module *Module) generateNestedFieldConverterStr(method pgs.Method, ix *importTracker, outputName string, edgeNumbers []protopack.Number, msg pgs.Message, varName string) (*string, error) {
+	converter := ""
+
+	var lastField pgs.Field
+	next := edgeNumbers[0]
+	deeper := edgeNumbers[1:]
+
+	// Generate the converter part for this level
+	converterPart, err := templateExecToString("field_message.tmpl", &messageFieldContext{
+		Number:     next,
+		InputName:  varName,
+		OutputName: varName,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Find the next message
+	for _, f := range msg.Fields() {
+		if next == protopack.Number(f.Descriptor().GetNumber()) {
+			lastField = f
+			break
+		}
+	}
+	var converterSubstring *string
+	if len(edgeNumbers) == 1 {
+		// Base case
+		paramContext, err := module.generateFieldConverter(method, next, lastField, ix, "", "value.String()", outputName)
```

Comment:
Will this work if the base case is a number?

Reply:
Value is a protoreflect value which I need to convert to a string. Depending on the type of the field it will use the corresponding tmpl file that will parse the string into the correct format.

For example:
[`{{.OutputName}}tmp, err := strconv.ParseInt({{.InputName}}, 10, 64)`](https://github.com/ductone/protoc-gen-apigw/blob/a9b8a49ccb45293dee1dd8216a9ec8097dd6f8b0/internal/apigw/templates/field_int.tmpl#LL3C1-L3C68)

Reply:
Sick